src/tools/launcher/util/launcher_util.cc

```diff
@@ -162,13 +162,14 @@ bool SetEnv(const string& env_name, const string& value) {
 string GetRandomStr(size_t len) {
   static const char alphabet[] =
       "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
-  ostringstream rand_str;
+  string rand_str;
```

Comment:
Can you add a comment here that using "string" instead of "ostringstream" is twice as fast?

Otherwise someone else might replace it again when refactoring this code in the future.

Reply:
I do have to add comments like this in a lot of places because Bazel seems to use `ostringstream` [everywhere](https://github.com/bazelbuild/bazel/blob/master/src/tools/launcher/java_launcher.cc#L194). Quite understandable since Bazel is more focus on its Java code.

I am planning to change many uses of `ostringstream` that are just concat `string`/`const char*`/`char` to just `string` `+=` operations. `ostringstream` (and `isotream` families) depend on `<locale>` library which is 150KB, I want to make the binary smaller.

It will great if we can use `absl::StrCat` from Abseil which is very efficient and can handle numbers as well (faster than `sprintf` and `ostringstream`).